test/apps/opeth.test.ts

```diff
@@ -0,0 +1,283 @@
+    // setup calculator
+    calculator = await MarginCalculator.new(addressBook.address)
+    // setup margin pool
+    marginPool = await MarginPool.new(addressBook.address)
+    // setup margin vault
+    const lib = await MarginVault.new()
+    // setup controllerProxy module
+    await Controller.link('MarginVault', lib.address)
+    controllerImplementation = await Controller.new(addressBook.address)
+    // setup mock Oracle module
+    oracle = await MockOracle.new(addressBook.address)
+    // setup whitelist module
+    whitelist = await Whitelist.new(addressBook.address)
+    await whitelist.whitelistCollateral(usdc.address)
+    await whitelist.whitelistCollateral(weth.address)
+    whitelist.whitelistProduct(weth.address, usdc.address, usdc.address, true)
+    whitelist.whitelistProduct(weth.address, usdc.address, weth.address, false)
+    // setup otoken
+    otokenImplementation = await Otoken.new()
+    // setup factory
+    otokenFactory = await OTokenFactory.new(addressBook.address)
+
+    // setup address book
+    await addressBook.setOracle(oracle.address)
+    await addressBook.setMarginCalculator(calculator.address)
+    await addressBook.setWhitelist(whitelist.address)
+    await addressBook.setMarginPool(marginPool.address)
+    await addressBook.setOtokenFactory(otokenFactory.address)
+    await addressBook.setOtokenImpl(otokenImplementation.address)
+    await addressBook.setController(controllerImplementation.address)
+
+    const controllerProxyAddress = await addressBook.getController()
+    controllerProxy = await Controller.at(controllerProxyAddress)
+
+    await otokenFactory.createOtoken(
+      weth.address,
+      usdc.address,
+      usdc.address,
+      createTokenAmount(strike, 8),
+      expiry,
+      true,
+    )
+
+    const putOtokenAddress = await otokenFactory.getOtoken(
+      weth.address,
+      usdc.address,
+      usdc.address,
+      createTokenAmount(strike, 8),
+      expiry,
+      true,
+    )
+
+    putOToken = await Otoken.at(putOtokenAddress)
+
+    // mint usdc to user
+    const accountOwner2Usdc = createTokenAmount(strike * optionsAmount, usdcDecimals)
+
+    await Promise.all([
+      usdc.mint(accountOwner2, accountOwner2Usdc),
+      usdc.approve(marginPool.address, accountOwner2Usdc, {from: accountOwner2}),
+    ])
+
+    const vaultCounter2Before = new BigNumber(await controllerProxy.getAccountVaultCounter(accountOwner2))
+    vaultCounter2 = vaultCounter2Before.toNumber() + 1
+
+    opeth = await Opeth.new()
+    await opeth.init(controllerProxyAddress, putOtokenAddress)
+  })
+
+  describe('Integration test: Opeth mint/redeem', () => {
+    const scaledOptionsAmount = createTokenAmount(optionsAmount, 8)
+    before('accountOwner2 mints the put option, sends it to accountOwner1.', async () => {
+      const collateralToMintLong = strike * optionsAmount
+      const scaledCollateralToMintLong = createTokenAmount(collateralToMintLong, usdcDecimals)
```

Comment:
```suggestion
      const scaledCollateralToMint = createTokenAmount(collateralToMintLong, usdcDecimals)
```
Just suggestion but I think the name `scaledCollateralToMint` is less confusing than `CollateralToMintLong`

Reply:
Fixed!